### PR TITLE
[3.3.5] Core/Spells: Fix wrong damage calculation for SPELL_AURA_PERIODIC_LEECH

### PR DESCRIPTION
**Pull request by Eliminationzx**
Closes https://github.com/TrinityCore/TrinityCore/issues/16495
Spells with aura attribute `SPELL_AURA_PERIODIC_LEECH` has negative damage calculation.
(For example spell [Devouring Plague](http://wotlk.openwow.com/spell=2944))

Damage should be calculated by `GetMaxHealth()` instead of `GetHealth()` in that case.

> ...GetHealth function will calculate currently amount of HP on target e.g. if target has 500/1000 hp it will calculate amount of 500 and if it's GetMaxHealth function, it will calculate maximal hp of target (1000 HP in this example)...

Thanks to @Goatform for detailed description.

### CODE REVIEW

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 1]
```diff
@@ -1301,8 +1301,7 @@ void Aura::HandleAuraSpecificMods(AuraApplication const* aurApp, Unit* caster, b
                     // Improved Devouring Plague
                     if (AuraEffect const* aurEff = caster->GetDummyAuraEffect(SPELLFAMILY_PRIEST, 3790, 1))
                     {
-                        uint32 damage = caster->SpellDamageBonusDone(target, GetSpellInfo(), GetEffect(0)->GetAmount(), DOT);
-                        damage *= caster->SpellDamagePctDone(target, GetSpellInfo(), SPELL_DIRECT_DAMAGE);
+                        uint32 damage = caster->SpellDamageBonusDone(target, GetSpellInfo(), GetEffect(0)->GetAmount(), DOT) * caster->SpellDamagePctDone(target, GetSpellInfo(), SPELL_DIRECT_DAMAGE);
```
Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: There is no change here - only code formatting changes should not be submitted in PRs that fix other things

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 2]
```diff
@@ -5998,8 +5998,8 @@ void AuraEffect::HandlePeriodicHealthLeechAuraTick(Unit* target, Unit* caster) c
 
     caster->CalcAbsorbResist(target, GetSpellInfo()->GetSchoolMask(), DOT, damage, &absorb, &resist, m_spellInfo);
 
-    if (target->GetHealth() < damage)
-        damage = uint32(target->GetHealth());
+    if (target->GetMaxHealth() < damage)
+        damage = target->GetMaxHealth();
```
Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: What is the result of this change? Calculating heal from all damage including overkill?

**Comment by jackpoz** [comment]
> Closes #16495

This does not really look like a proper PR description, you should describe in detail what behavior your code changes, why it does that and how it does that

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp (hunk 2 above)
Comment (Goatform) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: Well, GetHealth function will calculate currently amount of HP on target e.g. if target has 500/1000 hp it will calculate amount of 500 and if it's GetMaxHealth function, it will calculate maximal hp of target (1000 HP in this example).

File: src/server/game/Spells/Auras/SpellAuras.cpp (hunk 1 above)
Comment (Goatform) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: Yes I agree with @Shauren. This part should be reverted.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp (hunk 2 above)
Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: Yes, but why is this needed here?

**Comment by Eliminationzx** [comment]
Thank you for the feedback.

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp (hunk 2 above)
Comment (Goatform) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp]: I think that @Eliminationzx will need to answer on that question. From my aspect of looking as dev, this isn't very smart way because this could affect other things as well, at least in my opinion.

**Comment by Goatform** [comment]
@jackpoz, I think that this could be marked for Branch-6.x.  as well.

**Comment by Goatform** [comment]
@jackpoz, I think that this could be marked for Branch-6.x.  as well.

**Comment by Keader** [comment]
This PR Works for me :+1:

**Comment by Goatform** [comment]
@Keader, I already explained that.

**Comment by Shauren** [comment]
Rejecting due to insufficient description on how this fixes the issue or what the actual issue is.
Even if it has any visible effect it only masks the real issue (if one exist, I could not reproduce it given steps in #16495)
